mlflow/models/utils.py

```diff
@@ -289,10 +289,23 @@ def _enforce_tensor_spec(
     Enforce the input tensor shape and type matches the provided tensor spec.
     """
     expected_shape = tensor_spec.shape
+    expected_type = tensor_spec.type
     actual_shape = values.shape
-
     actual_type = values.dtype if isinstance(values, np.ndarray) else values.data.dtype
 
+    # This logic is for handling "ragged" arrays. The first check is for a standard numpy shape
+    # representation of a ragged array. The second is for handling a more manual specification
+    # of shape while support an input which is a ragged array.
```

Comment:
Can we include an example of each in the inline comment?

Reply:
Can we also add a section about ragged arrays to https://mlflow.org/docs/latest/models.html#signature-enforcement?

Reply:
Added a pair of inline samples and documentation. let me know what you think